docs/reference/extensions/env-injector-extension.rst

```diff
@@ -0,0 +1,126 @@
+.. _env-injector-extension:
+
+env-injector extension
+======================
+
+The env-injector extension, referred to internally as ``env-injector``, provides an
+interface for altering snap behavior with user-defined environment variables.
+
+When you add the extension to individual apps inside a snap, you open the way for the
+user to `configure the snap <https://snapcraft.io/docs/configuration-in-snaps>`_ or
+affect its behaviour with environment variables.
+
+
+How it works
+------------
+
+This extension splits environment variables into two types:
+
+- *Global* environment variables are visible to all apps in the snap that use
+  env-injector.
+- *Local* environment variables are visible only to apps with env-ibjector that the user
+  specifically targets.
+
+At a high level, the extension has the following processes and rules during build and
+runtime.
+
+
+Environment variable order of precedence
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Environment variables are processed in the following order:
+
+1. Variables extracted from the global ``.env`` file.
+2. Variables extracted from the local ``.env`` file.
+3. Variables from global snap config.
+4. Variables from local snap config.
+
+
+App build process
+~~~~~~~~~~~~~~~~~
+
+.. TODO: Put a link to exporter program repository
```

Comment:
Is it worth noting that this adds a part that builds the exporter?

Reply:
I'm not familiar enough to have a sense of it. I've already significantly rewritten the doc, so if we can reign in the scope...

Reply:
Gonna put this in the "let's wait until complaints" category then. It's an implementation detail, and while this is probably the correct page for that detail, I'm not sure it's worth surfacing. If people not knowing causes problems we can revisit at that point.